applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -150,6 +152,29 @@ impl wallet_server::Wallet for WalletGrpcServer {
         }))
     }
 
+    async fn set_base_node(
+        &self,
+        request: Request<SetBaseNodeRequest>,
+    ) -> Result<Response<SetBaseNodeResponse>, Status> {
+        let message = request.into_inner();
+        let public_key = PublicKey::from_bytes(message.public_key.as_slice())
```

Comment:
Based on @mikethetike's comment I realized this is the problem. This should actually be using from_hex(...).

Reply:
but base64 bytes also does the job

Reply:
it would be if it were `string` and not `bytes` in the proto file

Reply:
I see, will do it along with the commit signing. 😄

Reply:
Changed, so test as:
```
grpcurl -plaintext \
    -import-path ./applications/tari_app_grpc/proto \
    -proto wallet.proto \
    -d '{"public_key_hex": "1ed8d6e659bdabc5bd1f7cb6f8cfbbdbd97f5413fb8c40fc35fcc4cccb54df69", "net_address": "/ip4/127.0.0.1/tcp/18142"}' \
    localhost:18143 \
    tari.rpc.Wallet.SetBaseNode
```

Also learned to sign the commit.